tls/extensions/s2n_cert_authorities.h

```diff
@@ -19,6 +19,16 @@
 #include "tls/extensions/s2n_extension_type.h"
 #include "tls/s2n_connection.h"
 
+/* The only fixed bound on the size of the certificate_authorities is the maximum
```

Comment:
The limits seems reasonable/good, but this explanation doesn't entirely make sense to me. 

> but we need to ensure enough space for other extensions as well.

If each extension can be `UINT16_MAX` why does the size of the cert authorities extension affect the other extensions?

Reply:
Each extension can be up to UINT16_MAX because the extension size is a uint16_t, BUT all the extensions combined can also only be UINT16_MAX because the size of the extension list is only a uint16_t.

Reply:
Ahh, makes sense. Then maybe "maximum size of the extensions"?